WooCommerce/Classes/ViewRelated/Dashboard/Settings/Help/SupportForm/SupportFormViewModel.swift

```diff
@@ -156,6 +185,21 @@ extension SupportFormViewModel {
     }
 }
 
+// MARK: Private helpers
+private extension SupportFormViewModel {
+    func handleZendeskIdentityIfNeeded() {
+        guard !zendeskProvider.haveUserIdentity else {
+            DDLogDebug("Using existing Zendesk identity")
+            return
+        }
+
+        let identity = zendeskProvider.userSupportIdentity()
+        contactName = identity.name ?? ""
+        contactEmailAddress = identity.emailAddress ?? ""
+        shouldShowIdentityInput = true
+    }
```

Comment:
Here, if `contactName` and `contactEmailAddress` are both not null, can we pass them to `createIdentity()` to avoid having to show the alert?

Reply:
`contact` and `contactEmailAddress` can be non-nil if we get the info from credentials. You can confirm this in the implementation of`userSupportIdentity`. They are not the details that the user entered. If they are, we would have `zendeskProvider.haveUserIdentity` being true.

Reply:
I updated the method name in fa6bd29e23c0ed10d8eea37c9602dbaf2d6d4d94 to clarify this.

Reply:
> contact and contactEmailAddress can be non-nil if we get the info from credentials. You can confirm this in the implementation ofuserSupportIdentity

Yeah, I understand this. But why do users need / want to enter a different name/email than what's in their credentials?

With my suggestion above, the function algorithm can be:

```
    func handleZendeskIdentityIfNeeded() {
        let identity = zendeskProvider.userSupportIdentity()
        contactName = identity.name ?? ""
        contactEmailAddress = identity.emailAddress ?? ""

       // Add here: if contactName and ContactEmailAddress are not empty, then createIdentity() and return. 

        guard !zendeskProvider.haveUserIdentity else {
            DDLogDebug("Using existing Zendesk identity")
            return
        }

       // End of check. If still here, we need to request identity.
        shouldShowIdentityInput = true
    }
```

Reply:
I'll check.

Reply:
In android:

- it shows the alert if there's no `SUPPORT_EMAIL` value in AppPrefs ([code](https://github.com/woocommerce/woocommerce-android/blob/e54905d9e8c906ff293b6357a90fdd53e60e7d20/WooCommerce/src/main/kotlin/com/woocommerce/android/support/help/HelpActivity.kt#L143))
- `SUPPORT_EMAIL` is only filled after user fills in the alert ([here](https://github.com/woocommerce/woocommerce-android/blob/e54905d9e8c906ff293b6357a90fdd53e60e7d20/WooCommerce/src/main/kotlin/com/woocommerce/android/support/requests/SupportRequestFormViewModel.kt#L74) and [here](https://github.com/woocommerce/woocommerce-android/blob/e54905d9e8c906ff293b6357a90fdd53e60e7d20/WooCommerce/src/main/kotlin/com/woocommerce/android/support/help/HelpActivity.kt#L162).)
- `SUPPORT_EMAIL` is never filled automatically from credentials.

Conclusion: It's already the same logic in Android so feel free to ignore my solution.